pkg/controller/core/cohort_controller.go

```diff
@@ -104,7 +104,9 @@ func (r *CohortReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 
 	log.V(2).Info("Cohort is being created or updated", "resources", cohort.Spec.ResourceGroups)
-	r.cache.AddOrUpdateCohort(&cohort)
+	if err := r.cache.AddOrUpdateCohort(&cohort); err != nil {
+		return ctrl.Result{}, err
```

Comment:
Doesn't this cause a requeue into the reconciler?

Maybe we should log the issue, but then don't return the error.

Reply:
That makes sense, updated.